app/src/main/java/org/ostrya/presencepublisher/ForegroundService.java

```diff
@@ -193,8 +193,7 @@ private boolean isConnectedToWiFi() {
             return false;
         }
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
-            NetworkCapabilities networkCapabilities = connectivityManager.getNetworkCapabilities(connectivityManager.getActiveNetwork());
-            return networkCapabilities != null && networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI);
+            return wifiManager != null && wifiManager.getConnectionInfo().getNetworkId() != -1;
```

Comment:
I think I'd prefer getBSSID(), because it seems to be able to distinguish between no network connected and permission to access wifi info missing (see [doc](https://developer.android.com/reference/android/net/wifi/WifiInfo.html#getBSSID%28%29) ). Otherwise looks fine.

Reply:
Good point. What should happen if there is no permission?
Neither getNetworkId() nor getBSSID() seem to tell us if the device has a wifi connection when there is no permission to access location data:

State|getNetworkId()|getBSSID()
-----|----------------|----------
Wifi + permission|X|XX:XX:XX:XX:XX:XX
No wifi + permission|-1|null
Wifi + no permission|-1|02:00:00:00:00:00
No wifi + no permission|-1|02:00:00:00:00:00

Since the purpose of this function is to determine if there is a wifi connection (regardless of permissions), I now actually prefer the alternative solution of iterating over all networks to see if it includes a wifi network.

Reply:
If you want to go with the iteration, that's totally fine with me, but the app already shows a pop-up if necessary permissions (location in this case) are missing and states that if the permissions are missing, the app will not recognize the connectet network. So at the point of this check, I think assuming the permissions are set should be ok too.

Reply:
All 3 methods (getNetworkId, getBSSID and network iteration) solve the issue I was having so I'll let you make the final call.
Let me know which one you prefer and I'll update the pull request accordingly.

Reply:
Ok, after thinking about it, I agree with you that the iteration is the most fitting method for what this check is supposed to do. Thanks for your efforts!